crates/hyperqueue/src/server/job.rs

```diff
@@ -154,6 +149,15 @@ impl Job {
         }
     }
 
+    #[inline]
+    pub fn is_open(&self) -> bool {
+        self.is_open
+    }
+
+    pub fn close(&mut self) {
+        self.is_open = false;
```

Comment:
I meant that this should ideally call `self.check_completion`, so that it cannot be forgotten. Of course, it would then need to receive the `senders` and `now` parameters, not sure if that's worth it. Up to you :)